teps/0094-configuring-resources-at-runtime.md

```diff
@@ -0,0 +1,63 @@
+---
+status: proposed
+title: Configuring Resources at Runtime
+creation-date: '2021-11-08'
+last-updated: '2021-11-08'
+authors:
+- '@lbernick'
+---
+
+# TEP-0094: Configuring Resources at Runtime
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+<!-- /toc -->
+
+## Summary
+Add runtime configuration options for setting resource requirements of `Step`s and `Sidecar`s.
+
+Currently, users can specify resource requirements in a `Task` definition,
+via the `Resources` field of each `Step`, `StepTemplate`, or `Sidecar`. However, there is currently no support
+for modifying these requirements in a `TaskRun` or `PipelineTaskRun`.
```

Comment:
What does `PipelineTaskRun` mean here? does it refer to `PipelineTask`? Does it essentially mean modifying requirements in a `TaskRun` and/or `PipelineTask` with having reference to a catalog?

Or is it referring to `PipelineRun`? 🤔

Reply:
Just a run of a PipelineTask: https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L468-L481
The idea is to be able to change resource requirements for a Pipeline within a PipelineRun.

Reply:
For `PipelineTaskRun` - I'm guessing this will be the place in the API? https://tekton.dev/docs/pipelines/pipelineruns/#specifying-taskrunspecs

Reply:
yup!

Reply:
We could say in `TaskRun` (issued from PipelineRuns or one-shot) or something like this, but it's a nit.

Reply:
👍  to the nit, maybe for a future update - i think it's important to be cautious about introducing new terminology. "pipeline task" is already a bit confusing since this isn't a Tekton type, `PipelineTaskRun` looks like something I'd expect to be a Tekton type (esp mentioned alongside `TaskRun`)

Reply:
sounds good, I will change this in the subsequent pr.